### DISCUSSION TITLE
using-scrutor-to-automatically-register-your-services-with-the-asp-net-core-di-container/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#271] [Announcements]
# using-scrutor-to-automatically-register-your-services-with-the-asp-net-core-di-container/

In this post I describe how to use the open source library Scrutor to add assembly scanning capabilities to the ASP.NET Core DI container.

https://andrewlock.net/using-scrutor-to-automatically-register-your-services-with-the-asp-net-core-di-container/

[a comment by an automated account is not shown]

**Comment by ion-topala** [comment]
Thanks for the explanation of `ReplacementBehavior`, it helped me a lot.

**Comment by devFarshadFahimi** [comment]
Thanks for sharing this. it has a clear detail about Scrutor.
